osmosis-labs/osmosis
# migrate stXXX/XXX constant product pool -> stableswap pool

## What is the purpose of the change

Migrate pools 833, 817, and 810 from constant product -> stable swap.


## Brief Changelog

* add upgrade handler code
* test migration function
* test basic LP functionality (swap, withdraw after migration)


## Testing and Verifying

This change added tests and can be verified as follows:

- Added a unittest `TestMigrateBalancerToStablePools` to verify liquidity stays constant before/after the migration
- Added e2e tests
    - setup in `CreatePreUpgradeState` by LPing to the 3 pools
    - add `TestMigrateBalancerToStable` e2e test which swaps in and exits each pool

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? `CHANGELOG.md`
